crates/astria-composer/src/bundler.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    ds::RollupTxExt,
+    strategy::{NoStrategy, Strategy},
+};
+use color_eyre::eyre::{self, Context};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+use tokio::sync::mpsc as tokio_mpsc;
+
+/// The Bundler module bundles received transactions based on the various
+/// MEV strategies that are implemented.
+/// It then forwards the bundles to the Executor
+pub(crate) struct Bundler {
+    collector_receiver: UnboundedReceiver<RollupTxExt>,
+    executor_sender: UnboundedSender<Vec<RollupTxExt>>,
+}
+
+impl Bundler {
+    pub(crate) fn new(
+        collector_receiver: UnboundedReceiver<RollupTxExt>,
+    ) -> (Self, UnboundedReceiver<Vec<RollupTxExt>>) {
+        let (stream, sink) = tokio_mpsc::unbounded_channel::<Vec<RollupTxExt>>();
+        let new_strategy = Self {
+            collector_receiver,
+            executor_sender: stream,
+        };
+        (new_strategy, sink)
+    }
+
+    async fn bundle_transactions(
+        rollup_txs: Vec<RollupTxExt>,
+    ) -> Result<Vec<RollupTxExt>, eyre::Error> {
+        // NOTE: Replace this with arbitrary strategy logic
+        let bundle = NoStrategy::execute(rollup_txs).await?;
```

Comment:
should this be a generic for this function? like `async fn bundle_transactions<S: Strategy>` and then `S::execute()`? since I'm assuming this is the part we want to be switched out by users

Reply:
actually this function is private anyways so this might be unnecessary, but just curious about how you see this eventually being extended for different strategies

Reply:
Yeah this would make sense, but I think we should allow for cases where multiple strategies can be chained together. Plus, I don't think making this generic wouldn't actually reduce the amount of code that need to be replaced